src/SyntaxHighlighter/include/SyntaxHighlighter/Cpp.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SYNTAX_HIGHLIGHTER_CPP_H_
+#define SYNTAX_HIGHLIGHTER_CPP_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <qregularexpression.h>
+
+#include <QSyntaxHighlighter>
+
+namespace orbit_syntax_highlighter {
+
+/*
+  This a syntax highlighter for cpp.
+
+  It derives from QSyntaxHighlighter, so check out QSyntaxHighlighter's
+  documentation on how to use it. There are no additional settings or
+  APIs.
+*/
+class Cpp : public QSyntaxHighlighter {
+  Q_OBJECT
+  enum CppHighlighterState { kInitialState = 0, kOpenCommentState = 1, kOpenStringState = 2 };
```

Comment:
Do you need the explicit values? If not I would remove the `= 0`, `= 1`, and `= 2`.

Reply:
And it seems you don't need this enum for the API. So I would move it to the .cpp-file into a anonymous namespace.

Reply:
And could you make it an `enum class`. From the code I've seen, you already scope all usages properly.

Reply:
Done to the first 2. I didn't make an enum class just because I have to convert to an int for setCurrentBlockState. Is there a nice way to do that?

Reply:
No, not really. You have to call `static_cast` all the time.